set_choose.py

```diff
@@ -0,0 +1,28 @@
+import random
+from typing import TypeVar, Set
+T = TypeVar('T')
+
+
+#To note - the index reference is not constant
+#due to set
+def setChoice(s: Set[T]) -> T | None:
+    if len(s) == 0:
```

Comment:
```suggestion
def setChoice(s: Set[T]) -> T | None:
    """
    Chooses a single element at random from a set.
    When given the empty set returns None.
    When given a non-empty set always returns a value.
    """
    if len(s) == 0:
```

An example doc string. If this is applied, it will make it so hovering over the function name at point of use (in another file) will show the doc in the popup.

If you want to make a note about the implementation choice of a function it is common to do it as a comment after the docstring.